File: src/Atlasd/Battlenet/Protocols/Game/OldAuth.cs
using System;
using System.IO;

namespace Atlasd.Battlenet.Protocols.Game
{
    class OldAuth
    {
        public static byte[] CheckDoubleHashData(byte[] data, uint clientToken, uint serverToken)
        {
            var buf = new byte[28];
            using var m = new MemoryStream(buf);
            using var w = new BinaryWriter(m);

            w.Write(clientToken);
            w.Write(serverToken);
            w.Write(data);

            return MBNCSUtil.XSha1.CalculateHash(buf);
        }
    }
}

File: src/AtlasV/Battlenet/Protocols/Game/Other/OldAuth.vb
Imports System
Imports System.IO

Namespace AtlasV.Battlenet.Protocols.Game
    Class OldAuth
        Public Shared Function CheckDoubleHashData(ByVal varData As Byte(),
                                                   ByVal varClientToken As UInteger,
                                                   ByVal varServerToken As UInteger) As Byte()
            Dim buf = New Byte(27) {}

            Using m = New MemoryStream(buf)
                Using w = New BinaryWriter(m)
                    w.Write(varClientToken)
                    w.Write(varServerToken)
                    w.Write(varData)
                End Using
            End Using

            Return MBNCSUtil.XSha1.CalculateHash(buf)
        End Function
    End Class
End Namespace